Library/Homebrew/cmd/tap.rb

```diff
@@ -39,59 +38,23 @@ def install_tap user, repo
     true
   end
 
-  def link_tap_formula(paths, warn_about_conflicts=true)
-    ignores = (HOMEBREW_LIBRARY/"Formula/.gitignore").read.split rescue []
-    tapped = 0
-
-    paths.each do |path|
-      to = HOMEBREW_LIBRARY.join("Formula", path.basename)
-
-      # Unexpected, but possible, lets proceed as if nothing happened
-      to.delete if to.symlink? && to.resolved_path == path
-
-      begin
-        to.make_relative_symlink(path)
-      rescue SystemCallError
-        to = to.resolved_path if to.symlink?
-        opoo <<-EOS.undent if warn_about_conflicts
-          Could not create link for #{Tty.white}#{tap_ref(path)}#{Tty.reset}, as it
-          conflicts with #{Tty.white}#{tap_ref(to)}#{Tty.reset}. You will need to use the
-          fully-qualified name when referring this formula, e.g.
-            brew install #{tap_ref(path)}
-          EOS
-      else
-        ignores << path.basename.to_s
-        tapped += 1
-      end
+  def migrate_taps(autoskip=true)
+    # migrate taps from symlink based to directories based.
+    ignore = HOMEBREW_LIBRARY/"Formula/.gitignore"
+    if ignore.exist?
+      File.delete ignore
+    else
+      return if autoskip
```

Comment:
Using `HOMEBREW_LIBRARY/"Formula/.gitignore"` to determine whether we should migrate or not.